contracts/MetaIdentityManager.sol

```diff
@@ -0,0 +1,243 @@
+        _;
+    }
+
+    modifier onlyOlderOwner(address identity, address sender) {
+        require(isOlderOwner(identity, sender));
+        _;
+    }
+
+    modifier onlyRecovery(address identity, address sender) {
+        require(recoveryKeys[identity] == sender);
+        _;
+    }
+
+    modifier rateLimited(Proxy identity, address sender) {
+        require(limiter[identity][sender] < (now - adminRate));
+        limiter[identity][sender] = now;
+        _;
+    }
+
+    modifier validAddress(address addr) { //protects against some weird attacks
+        require(addr != address(0));
+        _;
+    }
+
+    /// @dev Contract constructor sets initial timelocks and meta-tx relay address
+    /// @param _userTimeLock Time before new owner added by recovery can control proxy
+    /// @param _adminTimeLock Time before new owner can add/remove owners
+    /// @param _adminRate Time period used for rate limiting a given key for admin functionality
+    /// @param _relayAddress Address of meta transaction relay contract
+    function MetaIdentityManager(uint _userTimeLock, uint _adminTimeLock, uint _adminRate, address _relayAddress) {
+        require(_adminTimeLock >= _userTimeLock);
+        adminTimeLock = _adminTimeLock;
+        userTimeLock = _userTimeLock;
+        adminRate = _adminRate;
+        relay = _relayAddress;
+    }
+
+    /// @dev Creates a new proxy contract for an owner and recovery
+    /// @param owner Key who can use this contract to control proxy. Given full power
+    /// @param recoveryKey Key of recovery network or address from seed to recovery proxy
+    /// Gas cost of ~300,000
+    function createIdentity(address owner, address recoveryKey) public validAddress(recoveryKey) {
+        Proxy identity = new Proxy();
+        owners[identity][owner] = now - adminTimeLock; // This is to ensure original owner has full power from day one
+        recoveryKeys[identity] = recoveryKey;
+        LogIdentityCreated(identity, msg.sender, owner,  recoveryKey);
+    }
+
+    /// @dev Creates a new proxy contract for an owner and recovery and allows an initial forward call which would be to set the registry in our case
+    /// @param owner Key who can use this contract to control proxy. Given full power
+    /// @param recoveryKey Key of recovery network or address from seed to recovery proxy
+    /// @param destination Address of contract to be called after proxy is created
+    /// @param data of function to be called at the destination contract
+    function createIdentityWithCall(address owner, address recoveryKey, address destination, bytes data) public validAddress(recoveryKey) {
+        Proxy identity = new Proxy();
+        owners[identity][owner] = now - adminTimeLock; // This is to ensure original owner has full power from day one
+        recoveryKeys[identity] = recoveryKey;
+        LogIdentityCreated(identity, msg.sender, owner,  recoveryKey);
+        identity.forward(destination, 0, data);
```

Comment:
Should it check for the presence of the destination before calling it?

Reply:
Why? Calling the zero address will have the same effect as calling any other random address that isn't used.